network/crl.go

```diff
@@ -232,8 +234,9 @@ func (c DefaultCRLClient) Fetch(url string, allowLocal bool) ([]byte, error) {
 
 // CRLCacheItem is combination of fetched+parsed+verified CRL with fetch time
 type CRLCacheItem struct {
-	Fetched time.Time // When this CRL was fetched and cached
-	CRL     pkix.CertificateList
+	Fetched             time.Time                            // When this CRL was fetched and cached
+	CRL                 pkix.CertificateList                 // Parsed CRL itself
```

Comment:
hm, why we store it as value and make several copy operations if `v.Client.Fetch(url, allowLocal)` returns pointer? can we store just pointer and avoid extra copies?

Reply:
plus, what about to encapsulate all this fields and use CRLCacheItem as interface, and initialize it with constructor?
because previously we initialized with `cacheItem := &CRLCacheItem{Fetched: time.Now(), CRL: *crl}`
now we prepare map for it, set nil for a field of CRL result and only after that create struct value with:
```
cacheItem := &CRLCacheItem{
		Fetched:             time.Now(),
		CRL:                 *crl,
		RevokedCertificates: revokedCertificates,
	}
```
But it's not `DefaultCRLVerifier` responsibility to know how CacheItem works. It should initialize it with CRL result and `CacheItem` prepare itself and his map as it wants. And just provide simple API how to create item and how to get info about revoked cert: `CacheItem.Get(cert *x509.Certificate)(revokedCert, ok)`

And it is encapsulates that it uses SerialNumber of certificate for map, not some hash or any other cert fields.

Reply:
Let's do it in next CRL-related PR? Gonna mention your comment in the task.